smarts/core/utils/episodes.py

```diff
@@ -91,43 +163,41 @@ def episodes(n):
     Acts similar to python's `range(n)` but yielding episode loggers.
     """
     col_width = 18
-    with tp.TableContext(
-        [
-            "Episode",
-            "Sim T / Wall T",
-            "Total Steps",
-            "Steps / Sec",
-            "Scenario Map",
-            "Scenario Routes",
-            "Mission (Hash)",
-            "Scores",
-        ],
-        width=col_width,
-        style="round",
-    ) as table:
-        for i in range(n):
-            e = EpisodeLog(i)
-            yield e
-
-            row = (
-                f"{e.index}/{n}",
-                f"{e.sim2wall_ratio:.2f}",
-                e.steps,
-                f"{e.steps_per_second:.2f}",
-                e.scenario_map[:col_width],
-                e.scenario_routes[:col_width],
-                e.mission_hash[:col_width],
-            )
+    with EpisodeLogs(col_width, n) as episode_logs:
+        for _ in range(n):
+            yield episode_logs.reset()
+        episode_logs.reset()
 
-            score_summaries = [
-                f"{score:.2f} - {agent}" for agent, score in e.scores.items()
-            ]
 
-            if len(score_summaries) == 0:
-                table(row + ("",))
-                continue
+@dataclass
+class Episodes:
```

Comment:
It would be better named as `class Counter`.

Reply:
I am not sure since the utility of it is not just counting.